metaflow/plugins/secrets/secrets_decorator.py

```diff
@@ -0,0 +1,241 @@
+import json
+import os
+import re
+from collections import defaultdict
+
+from metaflow.exception import MetaflowException
+from metaflow.decorators import StepDecorator
+from metaflow.plugins.secrets.aws_secrets_manager_secrets_provider import (
+    AwsSecretsManagerSecretsProvider,
+)
+from metaflow.plugins.secrets.inline_secrets_provider import InlineSecretsProvider
+
+SECRETS_BACKEND_PROVIDERS = {
```

Comment:
this is what i am saying to move to __init__.py for plugins. Create a `SECRETS_BACKEND_PROVIDERS_SPEC` and add those in there like:
```
("inline", ".secrets.inline_secrets_provider.InlineSecretsProvider")
...
```
and then add a TYPE (or whatever you want to call it to all the secrets providers and then add it here: https://github.com/Netflix/metaflow/blob/38bcd81800f19478324337c17c836665877bce8e/metaflow/extension_support/plugins.py#L175. 

It seems like a minuscule change that won't impact you much but would make this a lot cleaner and mean that I can develop things without having to do much change.

Reply:
That's fair.  To the uninitiated, (like me) - not quite minuscule (may be just small)... but it's done!  Please help take a look to make sure I did everything right re: plugin / extension format during your second pass.